### PR TITLE
UI fixes

### PR DESCRIPTION
**Pull request by daniellok-db**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/daniellok-db/mlflow/pull/10190?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10190/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10190
```

</p>
</details>

### What changes are proposed in this pull request?

Addressing post-merge feedback from #10088

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

Checked that the schema table looks the same as in #10088

https://github.com/mlflow/mlflow/assets/148037680/7088352d-32d0-48df-8add-42e7826b303b

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [x] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/server/js/src/model-registry/components/SchemaTable.test.tsx [hunk 1]
```diff
@@ -6,14 +6,26 @@
  */
 
 import React from 'react';
-import { shallow, render } from 'enzyme';
```
Comment (daniellok-db) [inline mlflow/server/js/src/model-registry/components/SchemaTable.test.tsx:9]: move avay from using enzyme

File: mlflow/server/js/src/model-registry/components/SchemaTable.tsx [hunk 2]
```diff
@@ -84,18 +86,7 @@ function formatColumnName(spec: ColumnSpec | TensorSpec): React.ReactElement {
...
-        padding: 8,
-        marginTop: 8,
-        marginBottom: 8,
-        whiteSpace: 'pre-wrap',
-      }}
-    >
-      {repr}
-    </pre>
-  );
+  return <pre css={schemaTableStyles.signatureCodeBlock}>{repr}</pre>;
```
Comment (daniellok-db) [inline mlflow/server/js/src/model-registry/components/SchemaTable.tsx:89]: use `css` instead of `style`

Comment (hubertzub-db) [reply mlflow/server/js/src/model-registry/components/SchemaTable.tsx:89]: 🎉

File: mlflow/server/js/src/model-registry/components/SchemaTable.test.tsx [hunk 3]
```diff
@@ -39,49 +51,50 @@ describe('SchemaTable', () => {
...
+    expect(container.querySelector('.outer-table table')).not.toBeNull();
+    expect(container.querySelector('.inner-table table')).toBeNull();
+    expect(container.innerHTML).toContain('Inputs');
+    expect(container.innerHTML).toContain('Outputs');
+    expect(container.innerHTML).toContain('Name');
+    expect(container.innerHTML).toContain('Type');
+    expect(container.innerHTML).not.toContain('column1');
+    expect(container.innerHTML).not.toContain('string');
+    expect(container.innerHTML).not.toContain('score1');
+    expect(container.innerHTML).not.toContain('long');
```
Comment (hubertzub-db) [inline mlflow/server/js/src/model-registry/components/SchemaTable.test.tsx:73]: Just a side note: recommended (more BDD) approach with RTL (contrary to Enzyme) is to use `screen` object imported from the same package as `render`, then build assertions like
```
expect(screen.getByRole('cell', { name: 'Inputs' })).toBeInTheDocument();
```
In general, we try to use accessible attributes (roles, names, labels) in RTL tests in order to make the test as high level (thus closer to real user interaction) as possible. However, this ☝️ is pretty valid here as well esp. given the fact we're just migrating 1:1 from `enzyme` :)

File: mlflow/server/js/src/model-registry/components/SchemaTable.test.tsx [hunk 4]
```diff
@@ -6,14 +6,26 @@
...
+    throw new Error("Couldn't find the row to click");
+  }
+  fireEvent(
+    rows[rowIndex],
+    new MouseEvent('click', {
+      bubbles: true,
+      cancelable: true,
+    }),
+  );
+}
```
Comment (hubertzub-db) [inline mlflow/server/js/src/model-registry/components/SchemaTable.test.tsx:26]: Did you try using
```
import userEvent from '@testing-library/user-event';
...
userEvent.click(rows[rowIndex]);
```
? 
It's a bit higher level function that should be automatically covering more edge cases (e.g. with bubbling etc) and be more convenient

**Review by hubertzub-db — APPROVED** [review]
LG! Left few comments to experiment with, however it's not a must

File: mlflow/server/js/src/model-registry/components/SchemaTable.test.tsx (hunk 3 above)
Comment (daniellok-db) [reply mlflow/server/js/src/model-registry/components/SchemaTable.test.tsx:69]: I'll definitely keep this in mind for the future! It looks like in order to migrate to the `screen` API we'd probably have to add quite a few `data-testid`/`role`s to the code (which would be a good thing to do, but perhaps in a future PR)

File: mlflow/server/js/src/model-registry/components/SchemaTable.test.tsx (hunk 4 above)
Comment (daniellok-db) [reply mlflow/server/js/src/model-registry/components/SchemaTable.test.tsx]: done! thanks for the pointer

**Pull request merged**